packages/kit-bg/src/states/jotai/atoms/hardware.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { EAtomNames } from '../atomNames';
+import { globalAtom } from '../utils';
+
+import type { IDeviceType } from '@onekeyfe/hd-core';
+
+export enum EHardwareUiStateAction {
+  DeviceChecking = 'DeviceChecking',
+  EnterPinOnDevice = 'EnterPinOnDevice',
+  ProcessLoading = 'ProcessLoading',
+
+  // ----------------------------------------------
+
+  REQUEST_PIN = 'ui-request_pin',
+  INVALID_PIN = 'ui-invalid_pin',
+  REQUEST_BUTTON = 'ui-button',
+  REQUEST_PASSPHRASE = 'ui-request_passphrase',
+  REQUEST_PASSPHRASE_ON_DEVICE = 'ui-request_passphrase_on_device',
+
+  CLOSE_UI_WINDOW = 'ui-close_window',
+
+  BLUETOOTH_PERMISSION = 'ui-bluetooth_permission',
+  LOCATION_PERMISSION = 'ui-location_permission',
+  LOCATION_SERVICE_PERMISSION = 'ui-location_service_permission',
+
+  FIRMWARE_PROGRESS = 'ui-firmware-progress',
+}
+
+export type IHardwareUiPayload = {
+  uiRequestType: string;
+  eventType: string;
+  // ----------------------------------------------
+  deviceType: IDeviceType;
+  deviceId: string;
+  connectId: string;
+  isBootLoaderMode?: boolean;
```

Comment:
```suggestion
  isBootloaderMode?: boolean;
```
https://en.wikipedia.org/wiki/Bootloader

Reply:
啥意思，这是要改什么

Reply:
变量名，bootloader 中间的 L 不用大写，是一个单词